api/saml/configuration.py

```diff
@@ -0,0 +1,376 @@
+                },
+            },
+            'security': {
+                'authnRequestsSigned': identity_provider.want_authn_requests_signed
+            }
+        }
+
+        if len(identity_provider.signing_certificates) == 1 and \
+                len(identity_provider.encryption_certificates) == 1 and \
+                identity_provider.signing_certificates[0] == identity_provider.encryption_certificates[0]:
+            onelogin_identity_provider['idp']['x509cert'] = identity_provider.signing_certificates[0]
+        else:
+            if len(identity_provider.signing_certificates) > 0:
+                if 'x509certMulti' not in onelogin_identity_provider['idp']:
+                    onelogin_identity_provider['idp']['x509certMulti'] = {}
+
+                onelogin_identity_provider['idp']['x509certMulti']['signing'] = \
+                    identity_provider.signing_certificates
+            if len(identity_provider.encryption_certificates) > 0:
+                if 'x509certMulti' not in onelogin_identity_provider['idp']:
+                    onelogin_identity_provider['idp']['x509certMulti'] = {}
+
+                onelogin_identity_provider['idp']['x509certMulti']['encryption'] = \
+                    identity_provider.encryption_certificates
+
+        return onelogin_identity_provider
+
+    def _get_service_provider_settings(self, service_provider):
+        """Converts ServiceProviderMetadata object to the OneLogin's SAML Toolkit format
+
+        :param service_provider: ServiceProviderMetadata object
+        :type service_provider: ServiceProviderMetadata
+
+        :return: Dictionary containing service provider's settings in the OneLogin's SAML Toolkit format
+        :rtype: Dict
+        """
+        onelogin_service_provider = {
+            'sp': {
+                'entityId': service_provider.entity_id,
+                'assertionConsumerService': {
+                    'url': service_provider.acs_service.url,
+                    'binding': service_provider.acs_service.binding.value
+                },
+                'NameIDFormat': service_provider.name_id_format,
+                'x509cert': service_provider.certificate if service_provider.certificate else '',
+                'privateKey': service_provider.private_key if service_provider.private_key else ''
+            }
+        }
+
+        return onelogin_service_provider
+
+    def get_identity_provider_settings(self, entity_id):
+        """Returns a dictionary containing identity provider's settings in a OneLogin's SAML Toolkit format
+
+        :return: Dictionary containing identity provider's settings in a OneLogin's SAML Toolkit format
+        :rtype: Dict
+        """
+        if entity_id in self._identity_providers:
+            return self._identity_providers[entity_id]
+
+        identity_providers = filter(lambda idp: idp.entity_id == entity_id, self._configuration.identity_providers)
```

Comment:
This is cosmetic, but a list comprehension is more readable in situations like this:

`identity_providers = [idp for idp in self._configuration.identity_providers if idp.entity_id == entity_id]`

Reply:
Changed it to using a list comprehension